File: 209-minimum-size-subarray-sum/209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        beg, N = 0, len(nums)
        total, leng = 0, math.inf
        
        for end in range(N):
            total += nums[end]
            while total >= target:
                leng = min(leng, end-beg+1)
                total -= nums[beg]
                beg += 1
                        
        return leng if leng != math.inf else 0

File: 209-minimum-size-subarray-sum/209-minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int n = nums.size(), start = 0;
        int cur = 0, len = INT_MAX;
        
        for (int end = 0; end < n; end++) {
            cur += nums[end];
            while (cur - nums[start] >= target) {
                cur -= nums[start];
                start++;
            }
            if (cur >= target)
                len = min(len, end - start + 1);
        }
        return (len == INT_MAX? 0: len);
    }
};